api/datastore/mock.go

```diff
@@ -75,33 +76,42 @@ func (s sortA) Len() int           { return len(s) }
 func (s sortA) Less(i, j int) bool { return strings.Compare(s[i].Name, s[j].Name) < 0 }
 func (s sortA) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 
-func (m *mock) GetApps(ctx context.Context, appFilter *models.AppFilter) ([]*models.App, error) {
+func (m *mock) GetApps(ctx context.Context, filter *models.AppFilter) (*models.AppList, error) {
 	// sort them all first for cursoring (this is for testing, n is small & mock is not concurrent..)
 	sort.Sort(sortA(m.Apps))
 
+	if filter.Cursor != "" {
+		s, err := base64.RawURLEncoding.DecodeString(filter.Cursor)
+		if err != nil {
+			return nil, err
+		}
+		logrus.Error(s)
+		filter.Cursor = string(s)
```

Comment:
You are modifyng a user-supplied parameter here - cursor shoud probably be a var from heron in